query-node/mappings/src/content/utils.ts

```diff
@@ -108,18 +108,37 @@ export interface IReadProtobufArgumentsWithAssets extends IReadProtobufArguments
 */
 export class PropertyChange<T> {
 
-  static newUnset<T>() {
+  static newUnset<T>(): PropertyChange<T> {
     return new PropertyChange<T>('unset')
   }
 
-  static newNoChange<T>() {
+  static newNoChange<T>(): PropertyChange<T> {
     return new PropertyChange<T>('nochange')
   }
 
-  static newChange<T>(value: T) {
+  static newChange<T>(value: T): PropertyChange<T> {
     return new PropertyChange<T>('change', value)
   }
 
+  /*
+    Determines property change from the given object property.
+  */
+  static fromObjectProperty<
+    T,
+    Key extends string,
+    ChangedObject extends {[key in Key]?: T}
+  >(object: ChangedObject, key: Key): PropertyChange<T> {
+    if (!(key in object)) {
```

Comment:
Protobuf's `has*` checks are useful to some extent, but `PropertyChange` has been created because our logic behind updating/unsetting/keeping unchanged is more complex, and also, some fields we might be updating are not coming from metadata. 

The class was originally created to help with handling redundant content info like `avatarPhotoDataObject`, `avatarPhotoUrls`, `avatarPhotoAvailability` that are derived from metadata and runtime event parameters, then it got useful in few more places.